src/qibo/models/dbi/double_bracket.py

```diff
@@ -20,6 +22,17 @@ class DoubleBracketGeneratorType(Enum):
     # TODO: add double commutator (does it converge?)
 
 
+class DoubleBracketScheduling(Enum):
```

Comment:
I'm sorry but I don't think I understand exactly what you mean. Does that mean I should:
1. move the 3 scheduling functions to `utils.py` and call them from  `DoubleBracketIteration`
2. instead of using a separate class like `DoubleBracketGeneratorType(Enum)`, make the attribute strings? i.e. `scheduling="hyperopt"`?
Sorry if I misunderstood, please correct me if I'm wrong.

Reply:
No worries, what I was suggesting is the following:

1. Move the 3 scheduling functions to `utils.py` (or if you want create a dedicated file),
2.  rewrite the class `DoubleBracketScheduling` such as
```py
class DoubleBracketScheduling(Enum):

    hyperopt =  hyperopt_step
    grid_search = grid_search_step
    polynomial_approximation = polynomial_step

```
where `polynomial_step`, `hyperopt_step`, `grid_search_step` are the functions of the point 1
3. the attribute `scheduling` remains a `DoubleBracketScheduling`, and in the method `choose_step` you could avoid all the branching since the optimization method is called like 
```py 
self.scheduling(...)
```

Reply:
Thanks for the clarification. The steps are clear and have been implemented.